var/spack/repos/builtin/packages/oce/package.py

```diff
@@ -87,7 +87,11 @@ def install(self, spec, prefix):
                 '-DOCE_OSX_USE_COCOA:BOOL=ON',
             ])
 
-        options.append('-DCMAKE_INSTALL_NAME_DIR:PATH=%s/lib' % prefix)
+        if '.'.join(platform.mac_ver()[0].split('.')[:2]) == '10.12':
+            # use @rpath on Sierra due to limit of dynamic loader
+            options.append('-DCMAKE_MACOSX_RPATH=ON')
```

Comment:
@tgamblin @mathstuf @alalazo :  i don't know all the details of how `rpath` works on `linux` and `macos`, but from my short research on the topic, i would make `DCMAKE_MACOSX_RPATH=ON` default in `std_cmake_args` on `macOS`.